x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -687,25 +691,40 @@ private DomainConfig getDomainConfigForSubject(Subject subject) {
         }
     }
 
-    private void updateProfileForActivate(Subject subject, VersionedDocument versionedDocument, ActionListener<Profile> listener)
+    // package private for testing
+    void updateProfileForActivate(Subject subject, VersionedDocument currentVersionedDocument, ActionListener<Profile> listener)
         throws IOException {
-        final ProfileDocument profileDocument = updateWithSubject(versionedDocument.doc, subject);
+        final ProfileDocument newProfileDocument = updateWithSubject(currentVersionedDocument.doc, subject);
+        // If the profile content does not change and it is recently updated within last 30 seconds, do not update it again
+        // to avoid potential excessive version conflicts
+        if (newProfileDocument.user().equals(currentVersionedDocument.doc.user())
+            && currentVersionedDocument.doc.enabled()
+            && newProfileDocument.lastSynchronized() - currentVersionedDocument.doc.lastSynchronized() < ACTIVATE_INTERVAL_IN_MS) {
+            logger.debug(
+                "skip user profile activate update because last_synchronized [{}] is within grace period",
+                currentVersionedDocument.doc.lastSynchronized()
+            );
+            listener.onResponse(currentVersionedDocument.toProfile(Set.of()));
+            return;
+        }
 
         doUpdate(
             buildUpdateRequest(
-                profileDocument.uid(),
-                wrapProfileDocumentWithoutApplicationData(profileDocument),
-                RefreshPolicy.WAIT_UNTIL,
-                versionedDocument.primaryTerm,
-                versionedDocument.seqNo
+                newProfileDocument.uid(),
+                wrapProfileDocumentWithoutApplicationData(newProfileDocument),
+                RefreshPolicy.WAIT_UNTIL
             ),
```

Comment:
No behaviour change here. The existing code does not fetch `primaryTerm` and `seqNo` for search result. Hence the net effect is that the update request just runs without a particular expectation for `primaryTerm` and `seqNo`. This is desirable because activate should just rely on the regular update mechanism instead of tying to any specific version. The change here is to make the intention explicit.